docs/tutorial_build_automation.rst

```diff
@@ -155,10 +153,8 @@ Finally, here is the generated run script that ReFrame uses to run the test, onc
 .. code-block:: bash
 
    #!/bin/bash
-   . "$(spack location --spack-root)/share/spack/setup-env.sh"
    spack env create -d rfm_spack_env
-   spack env activate -V -d rfm_spack_env
-   spack load bzip2@1.0.6
+   eval `spack -e rfm_spack_env load --sh bzip2@1.0.6`
```

Comment:
I think this is reasonable; either you use `spack env activate`, or you use `spack load`, but not both, cause `spack env activate --sh` also returns shell-environment variables changes for the spack-environment root specs and their transient run/link type dependencies.

so, it's either `eval "$(spack -e [env name] load --sh [spec name])"` where you use the environment to pick the just-built bzip2, or assuming you only have one root spec in the environment, you do `eval "$(spack env activate --sh [env name])"`

Reply:
You can also put `concretize: together` instead of the default `concretize: separately` in the environment, and then you can be sure that `spack env activate --sh ...` always works (cant have multiple bzip2's in one env then)

Reply:
not sure if you can change the value of `spack:concretize: ...` from the command line though :(

Reply:
Users can pass a custom environment through the `self.build_system.environment` option and reframe will not auto-generate the environment if they do.